### PR TITLE
ENT supported storage updates

### PR DESCRIPTION
**Pull request by ccapurso**
A change was added to the `vault server` command in 1.12 that would prevent startup (when not in dev mode) if configured to use an unsupported storage backend. HashiCorp currently supports Consul (consul) and Integrated Storage (raft). We are modifying this change so that it will instead log a warning. This PR includes the associated documentation changes to the account for the change in startup failure to warning.

This PR is currently a draft to ensure that it is not merged prior to the 1.12.2 release.

### CODE REVIEW

**Comment by mladlow** [comment]
I think the language updates are good, but that we probably need to add a note about the differences in behavior between 1.12.0 & 1 vs subsequent 1.12s

**Comment by ccapurso** [comment]
> I think the language updates are good, but that we probably need to add a note about the differences in behavior between 1.12.0 & 1 vs subsequent 1.12s

Great point, thanks! I will add a callout in the partial which will get included in the upgrade guide for 1.12.x and the storage docs directly.

**Review by mladlow — APPROVED** [review]

File: website/content/partials/ent-supported-storage.mdx [hunk 1]
```diff
@@ -1,3 +1,4 @@
-For enterprise customers, HashiCorp provides official support for Consul and Vault's Integrated Storage as storage backends. Vault Enterprise will
-no longer startup if configured to use a storage backend other than Consul or Integrated Storage. This is meant to protect against issues caused by using
-unsupported backends that do not support transactional storage.
+For enterprise customers, HashiCorp provides official support for Vault's Integrated Storage and Consul as storage backends. Vault Enterprise customers are strongly
+recommended to use these supported storage backends for best outcomes. Version 1.12.0 of Vault Enterprise will not startup if configured to use a storage backend
```
Comment (mpalmi) [inline website/content/partials/ent-supported-storage.mdx:2]: nit: should "startup" be "start up"?

Comment (ccapurso) [reply website/content/partials/ent-supported-storage.mdx:2]: Yes, but I actually think it reads better if we just remove the word up. What do you think?

Comment (mpalmi) [reply website/content/partials/ent-supported-storage.mdx:2]: Even better! 👍

Comment (ccapurso) [reply website/content/partials/ent-supported-storage.mdx]: Thanks for the feedback :)

**Pull request merged**